walkthroughs/howto-http-headers/README.md

```diff
@@ -0,0 +1,82 @@
+  7. `white-node`: A Cloud Map Node for `color.header-mesh.local` that discovers instances with the attribute `ECS_TASK_DEFINITION_FAMILY` set to `white`.
+  8. `black-node`: A Cloud Map Node for `color.header-mesh.local` that discovers instances with the attribute `ECS_TASK_DEFINITION_FAMILY` set to `black`.
+  
+* 7 Routes: 
+  1. `color-route-purple`: A route with a header match rule to match weather the header name `color_header` _does not_ exist.
+  2. `color-route-yellow`: A route with a header match rule to match weather the header name `color_header` _does_ exist.
+  3. `color-route-blue`: A route with a header match rule to match weather the header `color_header` has a value within the range `[100, 150)` (excludes 150).
+  4. `color-route-green`: A route with a header match rule to match weather the header name `color_header` has a value that matches the regex `redor.*`.
+  5. `color-route-red`: A route with a header match rule to match weather the header name `color_header` has a value that matches the prefix `redorgreen`.
+  6. `color-route-white`: A route with a header match rule to match weather the pseudo header name `:method` has a value that matches `GET`.
+  7. `color-route-black`: A route with a header match rule to match weather the pseudo header name `:scheme` has a value that matches the prefix `https`.
+
+## Setup
+
+1. Your account id:
+    ```
+    export AWS_ACCOUNT_ID=<your_account_id>
+    ```
+2. Your region:
+    ```
+    export AWS_DEFAULT_REGION=<i.e. us-west-2>
+    ```
+    
+3. The latest envoy image, see https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html
+   ```
+   export ENVOY_IMAGE=<i.e. 111345817488.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:v1.11.1.1-prod>
+   ```
+    
+4. Deploy the resources, save the public endpoint output for later (this will take about 5-10 minutes to complete):
+    ```
+    ./deploy.sh
+    ```
+
+## Update Route
+
+1. Check the current route configuration by viewing your routes in the app mesh console, within the app.yaml, or by using the CLI.
+For instance, to describe the blue route using the CLI:
+  ```
+  aws appmesh describe-route --mesh-name header-mesh --virtual-router-name color-router --route-name color-route-blue
+  ```
+
+2. For example routing a request with the _color_header_ header value set to _redorgreencolor_ will route to the green service (matching on regex _redor.*_) even though it also
+applies to the red service (matching on the prefix _redorgreen_). This is because the route priority attribute for the green route has higher priority than the red route. 
+To see this run the following command where endpoint refers to the output from the deploy script in the previous section:    
+  ```
+  curl --header "color_header: redorgreencolor" <endpoint>
+  ```
+
+2. You can update any route by updating the corresponding header match resources within the app.yaml file within the ColorRoute[Color] resources. 
```

Comment:
tbh. I'd prefer we supply some cli commands to run to mutate the mesh or some alternative .yaml files with route changes. Diving into the cloudformation is not very user-friendly.